onnxscript/ir/_core.py

```diff
@@ -1612,6 +1612,35 @@ def shape(self, value: Shape | None) -> None:
             return
         raise TypeError(f"Expected value to be a Shape or None, got '{type(value)}'")
 
+    def _compute_const_value(self) -> None:
+        """Evaluates Constant nodes and sets the const_value attribute of the Value."""
+        node = self.producer()
```

Comment:
From our discussion: I would add a strong clause saying this should be limited to Constant nodes only and maintain O(1) access